src/vlcPlayer.c

```diff
@@ -23,62 +23,116 @@
 
 #include "vlcPlayer.h"
 
-void reload_media()
+Vlc *vlc_instance_new()
 {
-   libvlc_media_player_set_media(vlc.media_player, vlc.media) ;pause_player() ;
+   return malloc (sizeof(Vlc)) ;
 }
 
 void init_vlc()
 {
-   vlc.inst = libvlc_new(0, NULL) ;
-   vlc.media_player = libvlc_media_player_new(vlc.inst) ;
+   vlc = vlc_instance_new() ;
+   vlc->inst = libvlc_new(0, NULL) ;
+   vlc->media_player = libvlc_media_player_new(vlc->inst) ;
 }
 
 void quit_vlc()
 {
    free(settings) ;
-   libvlc_media_player_release(vlc.media_player) ;
-   libvlc_release(vlc.inst) ;
+   libvlc_media_player_release(vlc->media_player) ;
+   libvlc_release(vlc->inst) ;
 }
```

Comment:
You should free all your allocated stuff and also the gslist with `g_slist_free()`, and the filenames with `g_free()`.